js/scripts.js

```diff
@@ -20,126 +20,133 @@ var newGameBtn = document.getElementById('js-newGameButton');
-			pickElem.style.display = 'none'; 
-			resultsElem.style.display = 'none'; 
-	} 
+function setGameElements() {
+	switch (gameState) {
+	case 'started':
+		newGameElem.style.display = 'none';
+		winnerTable.style.display = 'none'; //*
+		pickElem.style.display = 'block';
+		resultsElem.style.display = 'block';
+		break;
+
+	case 'ended':
+		winnerBoard.innerText = "Wygrał " + winnerName; //*
+		winnerTable.style.display = 'block'; //*
+		newGameBtn.innerText = 'Jeszcze raz';
+
+	case 'notStarted':
+
+	default:
+		newGameElem.style.display = 'block';
+		pickElem.style.display = 'none';
+		resultsElem.style.display = 'none';
+	}
 }
 
-
 function newGame() {
-	player.name = prompt('Graczu, wpisz swoje imię', 'imię gracza');
-	if (player.name) { // czy powyższy wiersz nie implikuje powstania tej zmiennej w kazdym wypadku ?
+	player.name = prompt('Graczu, wpisz swoje imię', 'Imię gracza');
+	if (player.name) {
 		player.score = 0;
 		computer.score = 0;
 		gameState = 'started';
-		winnerName = ''; //*
+		winnerName = ''; 
 		setGameElements();
-		playerNameElem.innerHTML = player.name; 
-		setGamePoints(); 
-	} 
+		playerNameElem.innerHTML = player.name;
+		setGamePoints();
+	}
 }
 
+function playerPick(playerPick) { // and computer also!
+	var computerPick = getComputerPick();
+
+	playerPickElem.innerHTML = playerPick;
+	computerPickElem.innerHTML = computerPick;
 
-function playerPick(playerPick) {  // and computer also!
-	var computerPick = getComputerPick(); 
-	
-	playerPickElem.innerHTML = playerPick; 
-	computerPickElem.innerHTML = computerPick; 
-	
 	checkRoundWinner(playerPick, computerPick);
 	checkGlobalWinner();
 }
 
-
-function getComputerPick() {  
-	return possiblePicks[Math.floor(Math.random()*3)]; 
+function getComputerPick() {
+	return possiblePicks[Math.floor(Math.random() * 3)];
 }
 
 
 function checkRoundWinner(playerPick, computerPick) {
-	playerResultElem.innerHTML = computerResultElem.innerHTML = '';
-	
-	var winnerIs = 'player'; // domyślny!
-	
-	if (playerPick == computerPick) { 
-	winnerIs = 'noone'; // remis 
-	} else if ( 
-		(computerPick == possiblePicks[0] && playerPick == possiblePicks[1]) || 
-		(computerPick == possiblePicks[1] && playerPick == possiblePicks[2]) || 
+	if (playerPick == computerPick) {
+		setRoundWinnerResults(null, null, true);
+	} else if (
+		(computerPick == possiblePicks[0] && playerPick == possiblePicks[1]) ||
+		(computerPick == possiblePicks[1] && playerPick == possiblePicks[2]) ||
 		(computerPick == possiblePicks[2] && playerPick == possiblePicks[0])) {
-		winnerIs = 'computer';
+		setRoundWinnerResults(player, playerResultElem);
+	} else {
+		setRoundWinnerResults(computer, computerResultElem);
+	}
+
+	return setGamePoints(); 
+}
+
+
+function setRoundWinnerResults(winner, winnerResultContainer, remis) {
+	playerResultElem.innerHTML = '';
+	computerResultElem.innerHTML = '';
+	if (remis === true){
+		computerResultElem.innerHTML = "Remis";
+		playerResultElem.innerHTML = "Remis";
```

Comment:
Hihhi... tak to wykminilismy na konsultacjach :). Ale nie booj - coś wymyslę wieczorem i poprawię :)